Source/ACE/Entity/Player.cs

```diff
@@ -1763,33 +1794,106 @@ public void HandleActionWieldItem(Container container, uint itemId, uint locatio
                             HandlePickupItem(container, itemGuid, location, PropertyInstanceId.Wielder);
                             return;
                         }
+
                         // Ok do you have this - if not bail
-                        if (GetInventoryItem(itemGuid) == null) return;
+                        var item = GetInventoryItem(itemGuid);
+                        if (item == null)
+                            return;
+                        if ((EquipMask)location > EquipMask.RingLeft)
+                        {
+                            // We are going into a weapon, wand or shield slot.
+                            switch ((EquipMask)location)
+                            {
+                                case EquipMask.MissileWeapon:
+                                    {
+                                        childLocation = 2;
+                                        placementId = 3;
+                                        break;
+                                    }
+                                case EquipMask.Shield:
+                                    {
+                                        childLocation = 3;
+                                        placementId = 6;
+                                        break;
+                                    }
+                                default:
+                                    {
+                                        placementId = 1;
+                                        childLocation = 1;
+                                        break;
+                                    }
+                            }
+                            Children.Add(new EquippedItem(itemGuid.Full, (EquipMask)location));
```

Comment:
There are actually some interesting questions here about ownership.  We're making the item landblock visible.  The item should be on the landblock (the landblock needs a lookup ability if someone goes to examine it), but ultimately shouldn't the player be responsible for it?  E.g. who is responsible as the item's parent (e.g. calling its "Act" function when it had work to do -- like using item mana).  Who is responsible for updating/tracking the item's motion?